SEED-platform/seed
# Shared Field Settings Page

#### Any background context you want to provide?
There was an error when going to the shared fields settings page in a users organization. 

#### What's this PR do?
The depth of the fix ballooned as soon as the code was touched. This commit cleans up several things:
* Update MappingData() to break out the fields in PropertyState vs Property vs TaxLotState vs TaxLot
* Remove the old BuildingSnapshot_to_BuildingSnapshot method used in merging in favor of MappingData() object.
* Migration for adding a shared_field_type field to the Column's table. Now that columns are assigned to organizations only, a field that is shared by the org can be specified on the Column object nicely.
* Replace the old /api/v2/columns list to be the /api/v2/columns/retrieve_all
* Add a query argument to /api/v2/columns to only return columns that have been mapped to.
* Return the database name of the column along with the unique name ('name'). The unique name gets prepended with tax_/property_ and/or appended with _extra
* Rework the created/updated fields to work the same as other fields that can exist in both PropertyStates and TaxLotStates
* Remove the INTERNAL column in the sharing settings page
* Remove old classes used for sharing (ExportableField and SharedBuildingField)
* Fix the tests

#### How should this be manually tested?
Unit tests, monkey tests. Also go the the organization settings page and add some fields to be shared publicly. Make sure they persist. The public API still is non-existent and will be part of another PR down the road.

#### What are the relevant tickets?
n/a
#### Screenshots (if appropriate)
#### Definition of Done:
